libraries/AP_Math/control.cpp

```diff
@@ -324,6 +324,38 @@ void shape_pos_vel_accel_xy(const Vector2p& pos_input, const Vector2f& vel_input
     shape_vel_accel_xy(vel_target, accel_input, vel, accel, accel_max, jerk_max, dt, limit_total_accel);
 }
 
+// limit_accel_xy limits the acceleration to prioritise acceleration perpendicular to the provided velocity vector.
+bool limit_accel_xy(const Vector2f& vel, Vector2f& accel, float accel_max)
+{
+    // check accel_max is defined
+    if ( !is_positive(accel_max) ) {
+        return false;
+    }
+    // limit acceleration to accel_max while prioritizing cross track acceleration
+    if (accel.length_squared() > sq(accel_max)) {
+        if (vel.is_zero()) {
+            // We do not have a direction of travel so do a simple vector length limit
+            accel.limit_length(accel_max);
+        } else {
+            // calculate acceleration in the direction of and perpendicular to the velocity input
+            const Vector2f vel_input_unit = vel.normalized();
+            // acceleration in the direction of travel
+            float accel_dir = vel_input_unit * accel;
+            // cross track acceleration
+            Vector2f accel_cross =  accel - (vel_input_unit * accel_dir);
+            if ( accel_cross.limit_length(accel_max) ) {
```

Comment:
nitpick: these white spaces within the brackets can go..

Reply:
Done